tests/run/cpp_classes_def.pyx

```diff
@@ -23,6 +24,9 @@ cdef cppclass RegularPolygon(Shape):
     float area() const:
         cdef double theta = pi / this.n
         return this.radius * this.radius * sin(theta) * cos(theta) * this.n
+    @cython.test_assert_path_exists('//SwitchCaseNode')  # optimized correctly (https://github.com/cython/cython/issues/4212)
+    int in_list(int x):
+        return x in [1,2,3]
```

Comment:
This is a somewhat misplaced test. We shouldn't be testing optimisations here. I think the `with` statement is a better candidate, being a generic syntax construct. It suffers from the same problem.

Reply:
Possibly... my thinking was that a) I'd rather test for the optimization rather the the `with nogil` because in my mind that's the "real" bug and the `with nogil:` statement is more of a "symptom" and b) that if I'd put it in the file that tests these optimimizations (maybe tests/run/switch.pyx?) then it'd need an extra c++-only file.

So I think this is the right test, but maybe in the wrong place. But there was in a reason it was there.

Reply:
I wasn't thinking of `with nogil` specifically – that feels like a different case also to me. But the `WithTransform` also inherits from `SkipDeclarations` and therefore suffers from the same bug. Thus my preference for it (being a required syntax construct) over the `SwitchTransform`, which is an optional optimisation. I think it's reasonable to test here that all supported syntax works also inside of C++ classes and their methods. But we shouldn't rely on a specific optimisation being enabled or not. If you want to test those in the context of a C++ class, then you're right, it would best be done in a separate test file.